### ISSUE TITLE
an option to store uploaded into `/tmp` folder

### ISSUE THREAD

**Issue by tomasc** [tomasc/simple_form_attachments#17]
By default the uploaded file is passed to the controller, who takes care storing it into database and marking it as temporary / permanent.

It might be worth to implement optional mechanism (service?) for the uploaded files to be stored in the `/tmp` folder and another one that let's one find and copy the files from the `/tmp` folder wherever they should be stored permanently.

This gem has been extracted from Modulor, where the temporary files assigned to modules need to be stored in the db to be previewed and stored in drafts, and then marked permanent when a page is published.

However in uploaders to standard records a simpler solution would be to store the uploaded files to `/tmp` folder and then copy to its permanent location (database, S3, file system) once the record is saved. This is useful especially when storing files into object storage such as S3 etc. as it would save ourselves unnecessary uploads to S3.

What do you think @asgerb ?

**Comment by asgerb** [comment]
I think this makes a lot of sense and would be a nice update. My only question would be why not to allow direct upload to S3, which unnecessary uploads would we avoid by storing it first in `/tmp`?

**Comment by tomasc** [comment]
You are right, although with storage in Mongo etc. it seems somehow cleaner to first store in `/tmp` (which the OS purges automatically) and then copy to persist.

This is not a priority, more of a 'nice to have'.

**Comment by tomasc** [comment]
@asgerb now I remembered why I wanted to do this … I have an asset repo and would like to allow a user to replace an image (asset). Uploading it to the datastore would immediately replace it, while one would like to do that after the owner document is saved.
